packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -358,36 +360,53 @@ function transform(root: IRElement, codeGen: CodeGen, state: State): t.Expressio
 
     function elementDataBag(element: IRElement): t.ObjectExpression {
         const data: t.Property[] = [];
-        const { classMap, className, style, styleMap, attrs, props, on, forKey, lwc } = element;
 
-        // Class attibute defined via string
-        if (className) {
-            const classExpression = bindExpression(className, element);
-            data.push(t.property(t.identifier('className'), classExpression));
-        }
-
-        // Class attribute defined via object
-        if (classMap) {
-            const classMapObj = objectToAST(classMap, () => t.literal(true));
-            data.push(t.property(t.identifier('classMap'), classMapObj));
-        }
-
-        // Style attribute defined via object
-        if (styleMap) {
-            const styleObj = objectToAST(styleMap, (key) => t.literal(styleMap[key]));
-            data.push(t.property(t.identifier('styleMap'), styleObj));
-        }
-
-        // Style attribute defined via string
-        if (style) {
-            const styleExpression = bindExpression(style, element);
-            data.push(t.property(t.identifier('style'), styleExpression));
-        }
+        const { attrs, props, on, forKey, lwc } = element;
 
         // Attributes
         if (attrs) {
-            const attrsObj = objectToAST(attrs, (key) => computeAttrValue(attrs[key], element));
-            data.push(t.property(t.identifier('attrs'), attrsObj));
+            const rest: { [name: string]: t.Expression } = {};
+
+            for (const [name, value] of Object.entries(attrs)) {
+                if (name === 'class') {
+                    // Handle class attribute:
+                    // - expression values are turned into a `className` property.
+                    // - string values are parsed and turned into a `classMap` object associating
+                    //   each individual class name with a `true` boolean.
```

Comment:
```suggestion
                    // Handle class attribute:
                    // - expression values are turned into a `className` property.
                    // - string values are parsed and turned into a `classMap` object associating
                    //   each individual class name with a `true` boolean.
                    // Examples:
                    //   `class={foo.bar}` becomes `{ className: $cmp.foo.bar }`
                    //   `class="foo bar"` becomes `{ classMap: { foo: true, bar: true } }`
```

Had to look at some fixtures before I understood what this was doing. 😛